src/main/java/org/isf/lab/gui/LabBrowser.java

```diff
@@ -202,64 +208,60 @@ private JButton getPrintTableButton() {
 		if (printTableButton == null) {
 			printTableButton = new JButton(MessageBundle.getMessage("angal.lab.printtable.btn"));
 			printTableButton.setMnemonic(MessageBundle.getMnemonic("angal.lab.printtable.btn.key"));
-			printTableButton.addActionListener(new ActionListener() {
+			printTableButton.addActionListener(arg0 -> {
+				typeSelected = ((Exam) comboExams.getSelectedItem()).toString();
+				if (typeSelected.equalsIgnoreCase(MessageBundle.getMessage("angal.common.all.txt"))) {
+					typeSelected = null;
+				}
 
-				public void actionPerformed(ActionEvent arg0) {
-					typeSelected = ((Exam) comboExams.getSelectedItem()).toString();
-					if (typeSelected.equalsIgnoreCase(MessageBundle.getMessage("angal.common.all.txt"))) {
-						typeSelected = null;
-					}
-					
-					try {
-						ArrayList<LaboratoryForPrint> labs;
-						labs = labManager.getLaboratoryForPrint(typeSelected, dateFrom.getDate(), dateTo.getDate());
-						if (!labs.isEmpty()) {
-							printManager.print(MessageBundle.getMessage("angal.common.laboratory.txt"), labs, 0);
-						}
-					} catch (OHServiceException e) {
-						OHServiceExceptionUtil.showMessages(e);
+				try {
+					ArrayList<LaboratoryForPrint> labs;
+					labs = labManager.getLaboratoryForPrint(typeSelected, dateFrom.getDate(), dateTo.getDate());
+					if (!labs.isEmpty()) {
+						printManager.print(MessageBundle.getMessage("angal.common.laboratory.txt"), labs, 0);
 					}
+				} catch (OHServiceException e) {
+					OHServiceExceptionUtil.showMessages(e);
 				}
 			});
 		}
 		return printTableButton;
 	}
-	
-	private JButton getPrintLabelButton(){
-		if (printLabelButton==null){
+
+	private JButton getPrintLabelButton() {
+		if (printLabelButton == null) {
 			printLabelButton = new JButton(MessageBundle.getMessage("angal.labnew.printlabel.btn"));
 			printLabelButton.setMnemonic(MessageBundle.getMnemonic("angal.labnew.printlabel.btn.key"));
-			printLabelButton.addActionListener(new ActionListener() {
-			
-				public void actionPerformed(ActionEvent arg0) {
-					Integer patId = null;
-					if (GeneralData.LABEXTENDED) {
-						selectedrow = jTable.getSelectedRow();
-						if (selectedrow < 0) {
-							int ok = MessageDialog.yesNoCancel(LabBrowser.this, "angal.lab.nopatientselectedprintempylabel.msg");
-							if (ok == JOptionPane.NO_OPTION) {
-								SelectPatient selectPatient = new SelectPatient(LabBrowser.this, false, true);
-								selectPatient.setVisible(true);
-								Patient patient = selectPatient.getPatient();
-								if (patient != null) {
-									patId = selectPatient.getPatient().getCode();
-								} else return;
-							}
-							if (ok == JOptionPane.CANCEL_OPTION) {
+			printLabelButton.addActionListener(arg0 -> {
```

Comment:
could we change all `arg0` and `e` parameters into `event`?

Reply:
The refactoring is using the information that was previously there: `public void actionPerformed(ActionEvent arg0) {`.   So yes it can be changed but it is a manual process looking for all occurrences.

Reply:
Yes I know, I meant just to perform a find/replace

Reply:
I will do that once I have converted the entire project to lambdas; would that be OK?

Reply:
sure, thanks!